src/visitable.cpp

```diff
@@ -831,8 +826,16 @@ long visitable<Character>::charges_of( const std::string &what, long limit ) con
 
     if( what == "UPS" ) {
         long qty = 0;
-        qty = sum_no_wrap( qty, charges_of( "UPS_off" ) );
-        qty = sum_no_wrap( qty, long( charges_of( "adv_UPS_off" ) / 0.6 ) );
+        if( p && p->has_item_with_flag( "UPS" ) ) {
+            for( auto &i : p->all_items_with_flag( "UPS" ) ) {
+                qty = sum_no_wrap( qty, charges_of( i->typeId() ) );
+            }
+        }
+        if( p && p->has_item_with_flag( "ADV_UPS" ) ) {
```

Comment:
You don't need to check whether there are any items with that flag, just iterate over them (as you do in the loop below). This is like
```C++
if( !some_container.empty() ) {
    for( elem : some_container ) {
        ...
    }
}
```
Loops handle empty ranges just fine. Same in the code above.